File: 106_touch/src/RaspberryPI/touch.py
# coding: utf-8
#
# FaBo Brick Sample
#
# brick_analog_touch
#

import RPi.GPIO as GPIO

LEDPIN = 4
TOUCHPIN = 7

GPIO.setwarnings(False)
GPIO.setmode( GPIO.BCM )
GPIO.setup( LEDPIN, GPIO.OUT )
GPIO.setup( TOUCHPIN, GPIO.IN )

while True:
     #圧力がかかっている間、LED点灯
    if( GPIO.input( TOUCHPIN ) ):
        GPIO.output( LEDPIN, False )
    else:
        GPIO.output( LEDPIN, True )

File: 0106_touch/src/Edison/touch.js
//
// FaBo Brick Sample
//
// #106 Touch Brick
//

//library
var m = require('mraa');

//pin setup
var myButton = new m.Gpio(14); //Touch sensor A0
var myLed    = new m.Gpio(2);  //LED D2

myButton.dir(m.DIR_IN);     //Touch sensor input
myLed.dir(m.DIR_OUT);       //LED output

//call loop function
loop();

function loop()
{

  if (myButton.read()){
    myLed.write(0);
  }
  else {
    myLed.write(1);
  }

  setTimeout(loop, 10);
}
